docs/design/guideline_recategorization.md

```diff
@@ -0,0 +1,178 @@
+
+- Mandatory guidelines - guidelines for which violation is never permitted.
+- Required guidelines - guidelines which can only be violated when supported by a deviation.
+- Advisory guidelines - recommendations for which violations are identified but are not required to be supported by a deviation.
+
+Guideline recategorization is possible by means of a Guideline Recategorization Plan (GRP). A GRP is a contract between the acquirer and supplier to determine how guidelines are applied.
+The GRP defines the additional category Disapplied to be used for Advisory guidelines which are to be ignored. Any other category can be recategorized into stricter categories to ensure that a guideline adheres to the associated policy.
+The following table summarizes the possible recategorizations.
+
+| Category  | Recategorizations               |
+| --------- | ------------------------------- |
+| Mandatory |                                 |
+| Required  | Mandatory                       |
+| Advisory  | Disapplied, Required, Mandatory |
+
+Other recategorizations, from here on denoted as invalid recategorizations, are not applied and are to be reported to the user.
+
+## Design
+
+Our design includes a Guideline Recategorization Plan specification, logic to apply the category policy to associated guidelines, and a SARIF result rewriter to reflect the new category in the results.
+The application of a policy will modify the behavior of a CodeQL queries implementing guidelines as follows:
+
+| Category   | Effect                                                               |
+| ---------- | -------------------------------------------------------------------- |
+| Mandatory  | Violations are reported, even if a deviation is applicable!          |
+| Required   | Violations are reported unless there exists an applicable deviation. |
+| Advisory   | Violations are reported unless there exists an applicable deviation. |
+| Disapplied | Violations are not reported.                                         |
+
+The SARIF rewriting will update the category of a guideline in a SARIF result file by updating the necessary tag information of a query.
+
+### Guideline Recategorization Plan specification
+
+The Guideline Recategorization Plan specification will build upon the configuration specification introduced for deviations by adding the additional primary section `guideline-recategorizations` to the `codeql-standards.yml` configuration file.
+The `guideline-recategorizations` section will be a series of compact mappings in YAML with the keys:
+
+- `rule-id` - the rule identifier that is recategorized.
+- `category` - the category assigned to the rule identified by rule-id
+
+Note: We specify the recategorization based on the rule-id instead of the query-id. This can be revised if feedback requires more fine-grained recategorization.
```

Comment:
This is a bit confusing, because it's outside the scope of what MISRA recategorization means, although as general mechanism perhaps it could be useful.  I think I would remove or clarify.

Reply:
This is a remnant of the proposal that I will remove for clarity.